features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -546,10 +546,20 @@ nsapi_error_t AT_CellularNetwork::set_registration(const char *plmn)
 
     if (!plmn) {
         tr_debug("Automatic network registration");
-        _at.cmd_start("AT+COPS=0");
+        uint8_t len=8;
+        uint8_t buf[8];
+        _at.cmd_start("AT+COPS?");
         _at.cmd_stop();
         _at.resp_start();
+        _at.read_bytes(buf,len);
         _at.resp_stop();
+        if( strncmp((char*)buf,"+COPS: 0",len)!=0) {
```

Comment:
Picky, I know, but this should be `if (strncmp(...`, i.e. the space is between the `if` and the opening bracket, not after the bracket.  You should also have a space either side of the `!=` in the `if()` statement; you don't want to give anyone any reason to question the PR :-).  

Otherwise, assuming it passed your tests, this all looks good to me.

Reply:
Thanks, will update in while.